teachertool/src/components/CatalogOverlay.tsx

```diff
@@ -0,0 +1,149 @@
+import { useContext, useMemo, useState } from "react";
+import { AppStateContext } from "../state/appStateContext";
+import { addCriteriaToRubric } from "../transforms/addCriteriaToRubric";
+import { CatalogCriteria } from "../types/criteria";
+import { getCatalogCriteria } from "../state/helpers";
+import { ReadOnlyCriteriaDisplay } from "./ReadonlyCriteriaDisplay";
+import { Strings } from "../constants";
+import { Button } from "react-common/components/controls/Button";
+import { getReadableCriteriaTemplate, makeToast } from "../utils";
+import { setCatalogOpen } from "../transforms/setCatalogOpen";
+import { classList } from "react-common/components/util";
+import { announceToScreenReader } from "../transforms/announceToScreenReader";
+import { FocusTrap } from "react-common/components/controls/FocusTrap";
+import css from "./styling/CatalogOverlay.module.scss";
+
+interface CatalogHeaderProps {
+    onClose: () => void;
+}
+const CatalogHeader: React.FC<CatalogHeaderProps> = ({ onClose }) => {
+    return (
+        <div className={css["header"]}>
+            <span className={css["title"]}>{Strings.SelectCriteriaDescription}</span>
+            <Button
+                className={css["close-button"]}
+                rightIcon="fas fa-times-circle"
+                onClick={onClose}
+                title={Strings.Close}
+            />
+        </div>
+    );
+};
+
+interface CatalogItemLabelProps {
+    catalogCriteria: CatalogCriteria;
+    allowsMultiple: boolean;
+    existingInstanceCount: number;
+    recentlyAdded: boolean;
+}
+const CatalogItemLabel: React.FC<CatalogItemLabelProps> = ({
+    catalogCriteria,
+    allowsMultiple,
+    existingInstanceCount,
+    recentlyAdded,
+}) => {
+    const canAddMore = allowsMultiple || existingInstanceCount === 0;
+    const showRecentlyAddedIndicator = recentlyAdded && canAddMore;
```

Comment:
In relation to the "adding multiple criteria in quick succession" problem, I'm wondering if this might also cause problems being static. It might be worth trying wrapping this in a `useEffect` to see if anything changes.

Reply:
I think moving these into a useEffect wouldn't help (and would actually be less efficient, since I'd have to make state vars and whatnot for them). They're just quick calculations based directly on the props. I think it's fine to re-do them. "adding multiple criteria in quick succession" should be fixed now with changes in c7b41141294206f7e462b8460e3774cf4d16f91a and 798107748afa97c45399c033c5630fbc2bb7940f